kythe/cxx/indexer/textproto/recordio_textparser.cc

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright 2023 The Kythe Authors. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <sstream>
+
+#include "absl/strings/ascii.h"
+#include "absl/strings/match.h"
+#include "absl/strings/string_view.h"
+#include "absl/types/optional.h"
+#include "glog/logging.h"
+#include "google/protobuf/descriptor.h"
+#include "google/protobuf/dynamic_message.h"
+#include "google/protobuf/text_format.h"
+
+namespace kythe {
+namespace lang_textproto {
+
+void ParseRecordioChunks(
+    std::string content, absl::optional<std::string> record_delimiter,
+    google::protobuf::DynamicMessageFactory& factory,
+    const google::protobuf::Descriptor* descriptor,
+    std::function<void(google::protobuf::Message*,
+                       google::protobuf::TextFormat::ParseInfoTree&, int)>
+        callback) {
+  std::string delimiter = "";
+  if (record_delimiter.has_value()) {
+    delimiter = absl::StripAsciiWhitespace(record_delimiter.value());
+  }
+  // TODO(chuthan): Accept parser as input instead of creating one here.
```

Comment:
Please unify this with the parser setup for the non-recordio case (e.g. handling this TODO).

Reply:
I've made this function to split and return the chunk only.. instead of also trying to parse.